addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -105,18 +105,35 @@ function insert(editor, data, isText = true) {
     let lastChildNode = false;
     if (fakeElLastChild.hasChildNodes()) {
         let tempCurrentNode = currentNode;
-        for (const n of [...fakeElLastChild.childNodes]) {
-            tempCurrentNode.after(n);
+        const children = [...fakeElLastChild.childNodes];
+        if (insertBefore) {
+            children.reverse();
+        }
+        for (const n of children) {
+            tempCurrentNode[insertBefore ? 'before' : 'after'](n);
             tempCurrentNode = n;
         }
-        lastChildNode = tempCurrentNode;
+        if (insertBefore) {
+            currentNode = tempCurrentNode;
+            lastChildNode = children[0];
+        } else {
+            lastChildNode = tempCurrentNode;
+        }
```

Comment:
I don't understand what is going on here. I'd like to discuss it together tomorrow. (I'd like you to rename the `n` variable to `child` anyway. I know it's not your code initially but since we're changing it anyway, now seems like a good time.)

Reply:
In one case we're inserting _after_, in the other _before_. So what constitutes the current and last node have to be different as the process is reversed.
Yeah I'll change that variable name, good call!